### PR TITLE
Feature/sort angles

### PR DESCRIPTION
**Pull request by dyang37**
We changed several things in the test script demo_sino_sort.py based on feedbacks from Prof. Bouman, Prof. Buzzard, and Jordan:

1. Changed the original angles range from [-90.90) deg to [0,180) deg. This is to make sure that the original and sorted sinograms are directly comparable. For the case where angles are from [-90.90), the sinogram view order after sorting actually corresponds to the original view order of [0,90),[-90,0) instead of [-90,90).

2. calculated transmission weights to test sino_sort for the weights array. Note that the weights are not used for actual reconstruction, since for the shepp-logan example using transmission weight will cause MBIR recon to never converge, while testing sino_sort on "unweighted" weights array does not make much sense since it is just an array of ones.

3. As requested by Prof. Buzzard, added print message to quantitatively compare the NRMSE between sinogram, weights, and reconstructions.

4. Added comparison of memory address of angles and weights before and after sorting.

5. Added diff image plots between three reconstructions.

6. Modified code based on Jordan's feedback.

7. Change variable names so that they make more sense.

Updated slides:
[Sorting_Sinogram.pptx](https://github.com/cabouman/svmbir/files/7298402/Sorting_Sinogram.pptx)

### CODE REVIEW

**Comment by cabouman** [comment]
Diyu, did you request reviews on this PR? If not, then please add them.

**Review by gbuzzard — COMMENTED** [review]
I thought we were no longer trying to create new memory.  Couldn't we just do 
    sino = sino[sorted_indices]
in place of line 51 and likewise for the update to weights?  
Then we would take out line 35.   
If we are trying to create new memory, then this is fine.

File: docs/source/overview.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ The super-voxel code can be 100x to 1000x faster than conventional code because
...
 
-*View Angle Ordering:* In order to achieve best reconstruction speed, we highly recommend views be sorted to be in monotonically increasing order on the inverval :math:`[0,2\pi]`. It is common for people ot collect tomography data using the "golden ratio" method. If this is done, than the view angles should first be wrapped modulo :math:`2\pi`, and then sorted to be in monotonically increasing order. Non-sequential ordering of views will distroy cache coherency and dramatically degrade ``svmbir`` performance.
+*View Angle Ordering:* In order to achieve best reconstruction speed, we highly recommend views be sorted to be in monotonically increasing order on the inverval :math:`[0,2\pi)`. This can be done using the function ``sino_sort``. It is common for people to collect tomography data using the "golden ratio" method. If this is done, than the view angles should first be wrapped modulo :math:`2\pi`, and then sorted to be in monotonically increasing order. Non-sequential ordering of views will distroy cache coherency and dramatically degrade ``svmbir`` performance.
```
Comment (gbuzzard) [inline docs/source/overview.rst:17]: Please change "than" to "then" in  "If this is done, than the view angles"

**Comment by cabouman** [comment]
Are these two statements definitely true?:

"The input arrays will remain unchanged."
"The sorted sino, angles, and weights arrays will be created in new memory addresses."

**Comment by sjkisner** [comment]
I noticed the demo.zip file added here contains image output and the __pycache__ folder, which I think was a mistake. It's a lot larger than it need to be.

**Comment by sjkisner** [comment]
In the demo there's a figure that's saved after the user is instructed to close it, so the file `output/angle_orders.png` is a blank image.

**Comment by cabouman** [comment]
Diyu, good catch. I added a bash script to create the demo.zip file. Can we easily purge the old demo.zip file from the repository? 

Also, can someone fix the problem with the python script that Jordan noticed?

**Pull request merged**